src/common/components/AnimatedModal/AnimatedModal.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { MouseEvent, useEffect, useRef, useState } from 'react'
+import { useKey } from 'react-use'
+import cn from 'classnames'
+import { X } from 'common/components/Icons'
+
+export type TAnimatedModalProps = {
+  open: boolean
+  onClose?(): void
+  render(): React.ReactNode
+  className?: string
+  closeButton?: boolean
+  easyToClose?: boolean // close by clicking on background or pressing Esc
+}
+
+const AnimatedModal = (props: TAnimatedModalProps): JSX.Element | null => {
+  // local open state is for closing animation
+  const [localOpen, setLocalOpen] = useState(props.open)
+
+  if (!props.open && !localOpen) {
+    return null
+  }
+
+  // logic is moved to separate component to init hooks, functions, variables only when AnimatedModal is open
+  return <AnimatedModalInner {...props} setLocalOpen={setLocalOpen} />
+}
+
+export default AnimatedModal
+
+const AnimatedModalInner = ({
+  open,
+  onClose,
+  closeButton,
+  easyToClose,
+  render,
+  className,
+  setLocalOpen,
+}: TAnimatedModalProps & { setLocalOpen(open: boolean): void }) => {
+  const modalRef = useRef<HTMLDivElement>()
+
+  useEffect(() => {
+    if (!open) {
+      const el = modalRef.current
+      if (!el) {
+        return
+      }
+
+      el.classList.remove('opacity-100')
+      el.addEventListener(
+        'transitionend',
+        () => {
+          setLocalOpen(false)
+        },
+        {
+          once: true,
+        },
+      )
+    }
+  }, [open])
+
+  useKey('Escape', () => easyToClose && onClose?.())
+
+  const setModalRef = (el: HTMLDivElement | null) => {
+    modalRef.current = el || undefined
+    if (el) {
+      setLocalOpen(true)
+      el.getBoundingClientRect()
+      el.classList.add('opacity-100')
+    }
+  }
```

Comment:
We can use Tailwind here.

Reply:
opacity-100 is a Tailwind class

Reply:
@CrazyOptimist , Sorry, I had to say "className". We can put this in className, I think.